src/mlang/optimizing_ir/partial_evaluation.ml

```diff
@@ -314,42 +314,32 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
                    (Pos.same_pos_as (Mir.Binop (op, new_e1, new_e2)) e1)
                    RegularFloat)
               (* first all the combinations giving undefined *)
-          | Mast.And, (Literal Undefined, _ | _, Undefined), _ ->
-              from_literal Undefined
-          | Mast.And, _, (Literal Undefined, _ | _, Undefined) ->
-              from_literal Undefined
-          | Mast.Mul, _, (Literal Undefined, _ | _, Undefined) ->
-              from_literal Undefined
-          | Mast.Mul, (Literal Undefined, _ | _, Undefined), _ ->
-              from_literal Undefined
-          | Mast.Div, (Literal Undefined, _ | _, Undefined), _ ->
-              from_literal Undefined
+          | Mast.And, (Literal Undefined, _ | _, Undefined), _
+          | Mast.And, _, (Literal Undefined, _ | _, Undefined)
+          | Mast.Mul, _, (Literal Undefined, _ | _, Undefined)
+          | Mast.Mul, (Literal Undefined, _ | _, Undefined), _
+          | Mast.Div, (Literal Undefined, _ | _, Undefined), _
           | Mast.Div, _, (Literal Undefined, _ | _, Undefined) ->
               from_literal Undefined
           (* logical or *)
-          | Mast.Or, (Literal Undefined, _ | _, Undefined), _ ->
-              (Pos.unmark new_e2, d2)
-          | Mast.Or, _, (Literal Undefined, _ | _, Undefined) ->
-              (Pos.unmark new_e1, d1)
-          | Mast.Or, (Literal (Float f), _), (_, Float) when f <> 0. ->
-              from_literal Mir.true_literal
-          | Mast.Or, (_, Float), (Literal (Float f), _) when f <> 0. ->
-              from_literal Mir.true_literal
+          | Mast.Or, (Literal Undefined, _ | _, Undefined), _
           | Mast.Or, (Literal (Float 0.), _), (_, Float) ->
               (Pos.unmark new_e2, d2)
+          | Mast.Or, _, (Literal Undefined, _ | _, Undefined)
           | Mast.Or, (_, Float), (Literal (Float 0.), _) ->
               (Pos.unmark new_e1, d1)
+          | Mast.Or, (Literal (Float _), _), _
+          | Mast.Or, _, (Literal (Float _), _) ->
+              from_literal Mir.true_literal
           (* logican and *)
-          | Mast.And, (Literal (Float 0.), _), (_, Float) ->
-              from_literal Mir.false_literal
-          | Mast.And, (_, Float), (Literal (Float 0.), _) ->
+          | Mast.And, (Literal (Float 0.), _), _
+          | Mast.And, _, (Literal (Float 0.), _) ->
               from_literal Mir.false_literal
-          | Mast.And, (Literal (Float f), _), (_, Float) when f <> 0. ->
-              (Pos.unmark new_e2, d2)
-          | Mast.And, (_, Float), (Literal (Float f), _) when f <> 0. ->
-              (Pos.unmark new_e1, d1)
```

Comment:
So all of this for `And` and `Or` is just code factorization right ? I don't see any semantic changes

Reply:
Yes